Minor improvements to SIL verifier

slavapestov opened the pull request:
Add some checks that we're not doing non-resilient things to resilient types inside inlinable functions, and fix two bugs caught by this.

slavapestov committed:
Sema: Resilient classes cannot have @inlinable deinits

slavapestov committed:
SIL: @objc thunks should not be serialized

slavapestov commented on test/ParseableInterface/ObjC.swiftinterface:
```diff
@@ -29,10 +29,9 @@ public class SomeClassInlinable {
-  @objc @inlinable deinit {
-    print("bye")
-  }
+  @objc deinit
```
@jrose-apple `@objc` is a no-op on a deinit isn't it? Should we warn and remove the attribute?

jrose-apple replied:
This is how it gets printed right now, so we have to handle reading it back in.

slavapestov replied:
Sure, so it would be a warning at best. I just think it might confuse people ("do I need to @objc my deinit if my class is @objc, or otherwise some weird edge case won't work?" "No, it actually does nothing")

jrose-apple replied:
\*shrug\* A warning sounds fine. I'd also be happy if we didn't print it here.

slavapestov commented:
@swift-ci Please smoke test

slavapestov commented:
@swift-ci Please test source compatibility

slavapestov committed:
SIL: Fix boxes test to actually emit SIB

slavapestov committed:
SIL: Verify that we don't emit function_refs to externally-defined methods in resilient classes

slavapestov committed:
SIL: Verifier checks resilience invariants for keypath instruction

slavapestov committed:
SIL: Verifier checks resilience invariants for global variable instructions

slavapestov committed:
SIL: Verifier checks resilience invariants for struct instructions

slavapestov committed:
SIL: Verifier checks resilience invariants for class instructions

slavapestov committed:
SIL: Fix oversight in switch_enum vs switch_enum_addr verification

Just like a switch_enum, switch_enum_addr must have a default case if
the enum is resilient from the current function.

slavapestov commented:
@swift-ci Please test

slavapestov commented:
@swift-ci Please test source compatibility

slavapestov commented:
Source compatibility “failures” are UPASS
